src/registrar/admin.py

```diff
@@ -50,13 +53,51 @@ class MyHostAdmin(AuditedAdmin):
     inlines = [HostIPInline]
 
 
+class DomainApplicationAdmin(AuditedAdmin):
+
+    """Customize the applications listing view."""
+
+    # Trigger action when a fieldset is changed
+    def save_model(self, request, obj, form, change):
```

Comment:
**question**: I can see where for a custom admin view, this is where state-change side effects can be made, but my concern is that it isn't very well integrated with the finite state machine that we use for application status (see https://github.com/cisagov/getgov/blob/main/docs/architecture/decisions/0015-use-django-fsm.md). Ideally, I would like to see this email sending code in an `investigate` transition method inside of `models/domain_application.py` like we do with `submit` (and ideally `withdraw` https://github.com/cisagov/getgov/issues/690). But I am not clear how that can integrate with the way that the admin allows directly editing the application's `status` field. Do we need to inspect the status change that was made in the admin view here and call the transition method from this method? Or is this a place where we can use something like `django-fsm-admin` to make the admin views aware of our transition methods? (https://github.com/gadventures/django-fsm-admin#usage)

Reply:
Fantastic feedback. On it.